Wiredcraft/Tools/WCRefresh.swift

```diff
@@ -0,0 +1,169 @@
+    ///   - header: header handle
+    ///   - footer: footer handle
+    /// - Returns:
+    public func refreshStatusBind(to scrollView: UIScrollView, _ header: (() -> Void)? = nil,  _ footer: (() -> Void)? = nil) -> Disposable {
+        
+        if header != nil {
+            scrollView.mj_header = MJRefreshNormalHeader {
+                scrollView.mj_footer?.endRefreshing()
+                header!()
+            }
+        }
+        
+        if footer != nil {
+            scrollView.mj_footer = MJRefreshAutoNormalFooter {
+                scrollView.mj_header?.endRefreshing()
+                footer!()
+            }
+        }
+        
+        /// refresh status
+        return refreshStatus.subscribe(onNext: { (status) in
+            switch status {
+            case .none:
+                scrollView.mj_footer?.isHidden = false
+                
+            case .beingHeaderRefresh:
+                scrollView.mj_header?.beginRefreshing()
+                
+            case .endHeaderRefresh:
+                scrollView.mj_header?.endRefreshing()
+                scrollView.mj_footer?.isHidden = false
+                
+            case .endFooterRefresh:
+                scrollView.mj_footer?.endRefreshing()
+                
+            case .endFooterRefreshWithNoMoreData:
+                scrollView.mj_footer?.endRefreshingWithNoMoreData()
+                
+            case .endAllRefresh:
+                scrollView.mj_header?.endRefreshing()
+                scrollView.mj_footer?.endRefreshing()
+                
+            case .beginFooterRefresh:
+                break
+            }
+        })
+    }
+}
+
+/*
+ * refresh and result handle
+ */
+public class WCRxRefresh: WCRefreshable {
+    
+    /// the data get from refresh action
+    public var data = PublishSubject<WCRefreshResponse>()
+    
+    /// refresh status
+    public var refreshStatus = BehaviorSubject(value: WCRefreshStatus.none)
+    
+    /// pull refresh
+    public var reload = PublishSubject<Void>()
+    
+    /// drag up load more
+    public var more = PublishSubject<Void>()
+    
+    /// dispose
+    let disposeBag = DisposeBag()
+    
+    public init() {
+        
+    }
+    
+    /// reset footer from no more data status to nomal
+    public func resetFooterStatus() {
+        self.refreshStatus.onNext(.endFooterRefresh)
+    }
+    
+    public func setNoMoreDataStstus() {
+        self.refreshStatus.onNext(.endFooterRefreshWithNoMoreData)
+    }
+    
+    /// set the network Observable
+    /// - Parameter observable: network request
+    public func set<T: WCRefreshResponse>(observable: Observable<T>?) {
+        guard let observable = observable else {
+            return
+        }
+    
+        self.reload = PublishSubject<Void>()
+        self.more = PublishSubject<Void>()
+        
+        //pull down regresh
+        self.reload.subscribe(onNext: { [weak self] (isReload) in
+            guard let `self` = self else {
```

Comment:
No question here, but just a tip. You don't need to use the ` `` `. You can just write `guard let self = self else`

Reply:
thank you for your tip